### PR TITLE
Add notarization manager unit tests & code fixes

### LINKED ISSUE #2245: Create tests for the commitments

### PR DESCRIPTION
**Pull request by jkrvivian**
# Description of change

Close #2245 #2283 #2284 

## Type of change

- Enhancement

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/notarization/manager_test.go [hunk 1]
```diff
@@ -9,195 +9,847 @@ import (
...
+
+		msg := testFramework.Message("Message2")
+		assert.Equal(t, epoch.Index(0), msg.EI())
+	}
+
+	assertExistenceOfBlock(t, testFramework, notarizationMgr, map[string]bool{
+		"Message1": true,
+	})
+
+	// Message3, issuing time epoch 3
```
Comment (daria305) [inline packages/notarization/manager_test.go:134]: We could start from here also checking if the reference to the `prevEC` is correct

**Review by daria305 — CHANGES_REQUESTED** [review]
Tests seems to cover nicely the flow and epoch changes :+1:  
I see we do not check committments itself, but it's fine as it is something we could do later with proofs. 
I don't like time.Sleep, not sure how hard it would be to ged rid of it

File: packages/notarization/manager_test.go [hunk 2]
```diff
@@ -370,16 +962,11 @@ func loadSnapshot(m *Manager, testFramework *tangle.MessageTestFramework) {
 	m.LoadSnapshot(snapshot)
 }
 
-func registerToTangleEvents(m *Manager, sfg *finality.SimpleFinalityGadget, testTangle *tangle.Tangle) {
+func registerToTangleEvents(sfg *finality.SimpleFinalityGadget, testTangle *tangle.Tangle) {
 	testTangle.ApprovalWeightManager.Events.MarkerWeightChanged.Hook(event.NewClosure(func(e *tangle.MarkerWeightChangedEvent) {
 		sfg.HandleMarker(e.Marker, e.Weight)
```
[the remark this answers is not in the document]
Comment (karimodm) [reply packages/notarization/manager_test.go:967]: The doggo really doesn't complain about these discarded return values; I also find it pretty ugly! Do we need it?

File: packages/epoch/types.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ func IndexFromTime(t time.Time) Index {
 		return 0
 	}
 
-	return Index(elapsedSeconds / int64(Duration))
+	return Index(elapsedSeconds / Duration)
 }
 
 // CurrentEI returns the EI at the current synced time.
```
[the remark this answers is not in the document]
Comment (karimodm) [reply packages/epoch/types.go:44]: True, but it seems like a useful function to keep around!

File: packages/notarization/manager_test.go [hunk 4]
```diff
@@ -9,195 +9,847 @@ import (
...
 
-func TestManager_PendingConflictsCount(t *testing.T) {
+func TestManager_pendingConflictsCounters(t *testing.T) {
 	testTangle := tangle.NewTestTangle()
-	m := NewManager(NewEpochManager(), NewEpochCommitmentFactory(testTangle.Options.Store, testTangle, 1), testTangle)
-	m.pendingConflictsCount[3] = 3
-	assert.Equal(t, uint64(3), m.PendingConflictsCount(3))
+	m := NewManager(NewEpochCommitmentFactory(testTangle.Options.Store, testTangle, 1), testTangle)
+	m.pendingConflictsCounters[3] = 3
+	assert.Equal(t, uint64(3), m.pendingConflictsCounters[3])
```
[the remark this answers is not in the document]
Comment (karimodm) [reply packages/notarization/manager_test.go:32]: You are right; also the NotNil is not needed as the assignation woul panic if nil

File: packages/notarization/manager_test.go [hunk 5]
```diff
@@ -208,146 +860,86 @@ func TestManager_UpdateStateMutationTree(t *testing.T) {
...
+func testNotarizationManager() (m *Manager) {
 	t := time.Now().Add(-25 * time.Minute).Unix()
 	testTangle := tangle.NewTestTangle()
 	interval := 5 * time.Minute
-	return NewManager(NewEpochManager(GenesisTime(t), Duration(interval)), NewEpochCommitmentFactory(testTangle.Options.Store, testTangle, 0), testTangle, MinCommittableEpochAge(10*time.Minute))
+	epoch.GenesisTime = t
+	epoch.Duration = int64(interval.Seconds())
+	m = NewManager(NewEpochCommitmentFactory(testTangle.Options.Store, testTangle, 0), testTangle, MinCommittableEpochAge(10*time.Minute))
+
+	m.epochCommitmentFactory.storage.SetLastCommittedEpochIndex(0)
```
[the remark this answers is not in the document]
Comment (karimodm) [reply packages/notarization/manager_test.go]: done

File: packages/notarization/manager_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkrvivian) [reply packages/notarization/manager_test.go:970]: I'll go for what it is now :grimacing:

[the remark this answers is not in the document]
Comment (daria305) [reply packages/notarization/manager_test.go:970]: For me it looks normal, as it indicates that you not handle error on purpose, and not just forgot about this. And my goland always underline unhandled errors

**Review by daria305 — APPROVED** [review]

**Pull request merged**